File: fe/lib/dio/response/GenerationResponses/MockGenerationResponse.dart

import 'package:ai_cv_generator/models/generation/CVData.dart';
import 'package:ai_cv_generator/models/user/UserModel.dart';
import 'package:json_annotation/json_annotation.dart';

part 'MockGenerationResponse.g.dart';

@JsonSerializable()
class MockGenerationResponse {
  UserModel mockgeneratedUser;
  CVData data;

  MockGenerationResponse({
    required this.mockgeneratedUser,
    required this.data
  });

  factory MockGenerationResponse.fromJson(Map<String, dynamic> json) => _$MockGenerationResponseFromJson(json);
  Map<String, dynamic> toJson() => _$MockGenerationResponseToJson(this);
}

File: ai-cv-generator-backend/src/main/java/com/revolvingSolutions/aicvgeneratorbackend/response/generation/MockGenerationResponse.java
package com.revolvingSolutions.aicvgeneratorbackend.response.generation;

import com.revolvingSolutions.aicvgeneratorbackend.model.aimodels.CVData;
import com.revolvingSolutions.aicvgeneratorbackend.model.user.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MockGenerationResponse {
    private User mockgeneratedUser;
    private CVData data;
}
